### PR TITLE
Token API docs

### PR DESCRIPTION
**Pull request by keithpitt**
I added a `/token` API to GET and DELETE the current token being used. I put together a page to document it.

I called it "Token Management", but perhaps "Management" is too much considering you can only do 1 thing? I thought "maybe in the future there'd be a "rotate" command or something, and that'd fit in here.

image

Changes welcome!

### CODE REVIEW

**Comment by toolmantim** [comment]
Woohoo!

I think just "Token API" for the page title, "Token" for the sidebar nav, and `/docs/apis/rest-api/token` for the URL? We need to update the app layout in the docs repo these days too (can't wait to remove this duplication!).

We should probably be clear that they're "API Access Tokens" in the words too (perhaps linking to https://buildkite.com/docs/apis/rest-api#authentication)… so as not to confuse them with other tokens (like agent tokens)?

**Comment by keithpitt** [comment]
@toolmantim cool cool - changes made!

**Comment by keithpitt** [comment]
Here's the URL: https://buildkite-docs-review-pr-488.herokuapp.com/docs/apis/rest-api/token

**Comment by toolmantim** [comment]
@harrietgrace I'm curious if you think "Token API" is obvious enough, or if not, perhaps "Access Token API" might be better? (i.e. "Access Token" in the nav, and "Access Token API" in the header? And perhaps move it to the top of the nav?)

🤷🏼‍♂️

**Comment by harrietgrace** [comment]
I think Token is alright by itself (rather than Access Token), but I think it needs to be 'Tokens' given that all the others are pluralized 😄 Otherwise looks great! We'll need to update this in bk/bk too given that the sidebar isn't actually coming from this repo.

**Comment by keithpitt** [comment]
@harrietgrace I left it as "Token" since there's only 1 token you can ever get information for (the current token). There's no "token list" - so never any plural.

Do you think it's OK with that in mind? Or is the pluralisation because the page is talking about "tokens" generally?

**Comment by harrietgrace** [comment]
I was just chatting about this with @toolmantim actually 😄I think it should be plural because as a user you can have multiple tokens. Also, because I would expect that a page called 'Token' or 'Access Token' would tell me about generating a token for the REST API and how to use it, I wouldn't expect an API page. 'Access Tokens' I would think was about managing my tokens/doing things with tokens, and even though the API only lets you deal with one at a time, I think it still makes sense to have it plural in the sidebar.

File: pages/apis/rest_api/token.md.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+}
+```
+
+Required scope: none
+
+Success response: `200 OK`
+
+## Revoke current API Access Token
+
+Immediately revokes the current [API Access Token](/docs/apis/rest-api#authentication) and sends an email to the owner. Once revoked, the token cannot be used anymore.
```
Comment (toolmantim) [inline pages/apis/rest_api/token.md.erb]:
```suggestion
Revokes the API Access Token that was used to make the API request. Once revoked, the API Access Token can no longer be used.
```

File: pages/apis/rest_api/token.md.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Token API
```
Comment (toolmantim) [inline pages/apis/rest_api/token.md.erb]:
```suggestion
# Access Token API
```

File: pages/apis/rest_api/token.md.erb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+<%= toc %>
```
Comment (toolmantim) [inline pages/apis/rest_api/token.md.erb]:
```suggestion
The Access Token API lets you perform actions on an individual [API Access Token](https://buildkite.com/docs/apis/rest-api#authentication), using the authentication details of REST API request.

<%= toc %>
```

File: pages/apis/rest_api/token.md.erb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+## Get current API Access Token
+
+Returns limited information about the current [API Access Token](/docs/apis/rest-api#authentication) being used to access the REST API.
```
Comment (toolmantim) [inline pages/apis/rest_api/token.md.erb]:
```suggestion
Returns details about the API Access Token that was used to make the API request.
```

**Comment by keithpitt** [comment]
@toolmantim happy for you to merge what ever changes you want in!

**Comment by toolmantim** [comment]
I'll make those docs changes!

Did you wanna make the REST endpoint `/v2/access-token` @keithpitt?

**Comment by toolmantim** [comment]
Based on Keith's demo words, we should probably add an intro for some of the common use cases of this API, such as finding an API Access Token and not knowing what user it's attached to, and wanting to make sure it's revoked.

**Pull request merged**